### PR TITLE
change alignment in attack code for more readability.

### PR DESCRIPTION
**Pull request by newfrenchy83**

### CODE REVIEW

File: src/units/attack_type.cpp [hunk 1]
```diff
@@ -101,17 +101,19 @@ attack_type::attack_type(const config& cfg) :
 		else
 			icon_ = "attacks/blank-attack.png";
 	}
-	if(cfg.has_attribute("alignment") && (cfg["alignment"] == "neutral" || cfg["alignment"] == "lawful" || cfg["alignment"] == "chaotic" || cfg["alignment"] == "liminal")){
-		alignment_str_ = cfg["alignment"].str();
-	} else if(self_){
-		alignment_str_ =unit_alignments::get_string(self_->alignment());
+	//if alignment_ don't exist but self_ is here(like when called in lua), made alignment_ equals to unit alignment.
+	if(!alignment_ && self_){
```
Comment (soliton-) [inline src/units/attack_type.cpp:105]: What is the point of considering `self_` here when it is accounted for in `attack_type::alignment()`?

**Comment by soliton-** [comment]
Apart from setting `alignment_` based on `self_` in the constructor this looks good.

**Comment by soliton-** [comment]
I wonder if a better strategy is to leave the fallback to the caller. Just return an unset optional if the weapon has no alignment and let the caller fallback to the unit's alignment. Then the caller can do the right thing even if an attack is somehow detached from a unit. Returning neutral leaves the caller no option to do whatever is best in their context since there is no way to know if it is the actual alignment or the fallback.

**Comment by newfrenchy83** [comment]
use optional cause bug in check , and i restested lua and fallback is ever neutral, i will simplify the code because lua solution don't work.

**Pull request closed without merging**